### PR TITLE
Add terraform basic

### PR DESCRIPTION
**Pull request by irenepapaspyratos**

### CODE REVIEW

File: scripts/prerequisites.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+S3_BUCKET_SRC=trady-cloud-src
+S3_BUCKET_TERRAFORM=trady-cloud-terraform
+REGION=us-west-2
+
+# Create buckets necessary for Terraform via script as the sandbox does not allow creating them via Terraform itself
+sh setup_buckets.sh $REGION $S3_BUCKET_SRC $S3_BUCKET_TERRAFORM 
+
+# Create from terraform-templates with variables: main.tf & variables.tf
+sed -e s/REGION/${REGION}/ -e s/S3_BUCKET_TERRAFORM/${S3_BUCKET_TERRAFORM}/  < ../infrastructure/templates/main.tf.template > ../infrastructure/main.tf
+sed s/REGION/${REGION}/ < ../infrastructure/templates/variables.tf.template > ../infrastructure/variables.tf
```
Comment (fabianschmauder) [inline scripts/prerequisites.sh]: do you really need this? 
You can also parse this parameters to terraform as configuration from your environment variables

File: scripts/setup_buckets.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Create bucket for resources in chosen region
+if aws s3 ls "s3://$2" 2>&1 | grep -q 'An error occurred.'
+then
+    aws s3api create-bucket --bucket $2 --region $1 --create-bucket-configuration LocationConstraint=$1
+    echo "Bucket '$2' has been created."
+else
+    echo "Bucket '$2' already exists."
+fi
```
Comment (fabianschmauder) [inline scripts/setup_buckets.sh]: I know we also did this. But you can also create script to create a bucket and parse the bucket name as parameter. 
Than you do not have this duplication for every bucket

File: infrastructure/templates/main.tf.template [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+
+   backend "s3" {
+    bucket = "S3_BUCKET_TERRAFORM"
+    key    = "terraform.tfstate"
+    region = "REGION"
```
Comment (fabianschmauder) [inline infrastructure/templates/main.tf.template]: you can set the region from outside
[config](https://developer.hashicorp.com/terraform/language/settings/backends/configuration)

`terraform init -input=false  -backend-config="bucket=irene-bucket"`

**Review by fabianschmauder — COMMENTED** [review]
Small hints :)

File: scripts/setup_buckets.sh (hunk 2 above)
Comment (irenepapaspyratos) [reply scripts/setup_buckets.sh]: 😲🤦‍♀️🙈

File: scripts/prerequisites.sh (hunk 1 above)
Comment (irenepapaspyratos) [reply scripts/prerequisites.sh]: I thought about maybe let the user later decide this via terminal input and wrote it like this to be as flexible as possible

File: scripts/functions_terraform.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+create_tf_variables() {
+    sed -e s/REGION/${1}/ -e s/S3_BUCKET_TERRAFORM/${2}/ \
+    < ../infrastructure/templates/variables.tf.template > ../infrastructure/variables.tf
+}
+
+# Create configuration file for s3 backend from template
+create_tf_backend_s3() {
+    sed -e s/REGION/${1}/ -e s/S3_BUCKET_TERRAFORM/${2}/ \
+    < ../infrastructure/templates/config.s3.tfbackend.template > ../infrastructure/config.s3.tfbackend
+}
```
Comment (fabianschmauder) [inline scripts/functions_terraform.sh:13]: I dont get why you need this. Maybe lets have a quick talk after the daily 🙈

File: scripts/prerequisites.sh (hunk 1 above)
Comment (fabianschmauder) [reply scripts/prerequisites.sh]: but you can do this with running 

`terraform init -input=false -backend-config="bucket=$TERRAFORM_STATE_BUCKET"`

Than you do not need to replace some thing

Also it is always better to define some thing when you do not need to configure it now

"convention over configuration" 

Helps you to reduce complexity in your code

File: scripts/functions_terraform.sh (hunk 4 above)
Comment (irenepapaspyratos) [reply scripts/functions_terraform.sh:13]: 😂 Ok. 🙃

File: scripts/prerequisites.sh (hunk 1 above)
Comment (irenepapaspyratos) [reply scripts/prerequisites.sh]: Whaaaaat? Are you trying to reduce my second name?!?! 🧐

File: scripts/functions_terraform.sh (hunk 4 above)
Comment (irenepapaspyratos) [reply scripts/functions_terraform.sh:13]: After our talk I will merge as is, but keep this branch to have a look at what you wrote at a later state. 😊

File: scripts/prerequisites.sh (hunk 1 above)
Comment (irenepapaspyratos) [reply scripts/prerequisites.sh]: See following suggestions

**Pull request merged**